website/pages/blog/setting-up-graphql-server-to-scale-with-graphql-codegen-server-preset.mdx

```diff
@@ -0,0 +1,435 @@
+
+```ts
+// Example: src/schema/user/resolvers/User.ts
+import type { UserResolvers } from './../../types.generated';
+export const WrongUser: UserResolvers = {
+  /* Implement User resolver logic here */
+};
+/* WARNING: The following resolver was missing from this file. Make sure it is properly implemented or there could be runtime errors. */
+export const User: UserResolvers = {
+  /* Implement User resolver logic here */
+};
+```
+
+#### Other GraphQL Types
+
+These other types are also supported by the preset:
+
+*   Union: A file is generated for every Union type.
+*   Scalars:
+    *   If the Scalar name matches one in [graphql-scalars](https://the-guild.dev/graphql/scalars), it is automatically imported from `graphql-scalars` into the resolver map so make sure to install it:
+    ```bash
+    yarn add graphql-scalars
+    ```
+    *   If the Scalar name does not exist in `graphql-scalars`, a file is generated for every Scalar type.
+
+For other currently non-supported types, we can declare them using the `externalResolvers` preset config.
+
+#### Mappers Convention
+
+Mappers can be added by exporting types or interfaces with `Mapper` suffixes from `.mappers.ts` files in each module. For example, `UserMapper` will be used as `User`'s mapper type.
+
+```ts
+// src/schema/user/schema.mappers.ts
+
+// This works! This will be used as mapper for `User` object type
+export { User as UserMapper } from 'external-module'
+
+// This 1 works! For `User1` object type
+export interface User1Mapper { 
+  id: string;
+}
+
+// This works 2! For `User2` object type
+export type User2Mapper = { id: string } 
+
+// This works 3! For `User3` object type
+interface User3Mapper {
+  id: string;
+}
+export { User3Mapper }
+```
+
+#### Gradual Migration Supported
+
+Do you have existing code and cannot migrate all in one big bang? The preset has `whitelistedModules` and `blacklistedModules` settings to support gradual migration of modules.
+
+#### Customisable Conventions
+
+All the above conventions are customisable! Check out the documentation for [more options](https://github.com/eddeee888/graphql-code-generator-plugins/tree/master/packages/typescript-resolver-files).
+
+## Summary
+
+In this blog post, we have explored the problems that can happen to a GraphQL server if it is not set up to scale such as ownership, enforcing best practices and minimising noise to Maintainers. We can use the server preset `@eddeee888/gcg-typescript-resolver-files` to solve these problems for GraphQL servers.
+
+At [SEEK](https://www.seek.com.au/), we are experimenting with this preset and getting positive feedback. [Let me know on Twitter](https://twitter.com/eddeee888) if it works for you too!
```

Comment:
Are we happy for me to call my current workplace here? 
I'll need to check if they are happy for me to call out here too

Reply:
Sure that is fine